app/llm/performance_enhanced.py

```
@@ -0,0 +1,142 @@
+            try:
+                if len(self._cache) >= self._max_size:
+                    self._cache.popitem(last=False)
+                self._cache[key] = value
+                self._cache.move_to_end(key)
+                return True
+            except Exception as e:
+                logger.error(f"Cache storage error: {e}")
+                return False
+
+    @property
+    def metrics(self) -> Dict[str, int]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)